server/monitor.go

```diff
@@ -2606,22 +2609,21 @@ func (s *Server) Jsz(opts *JSzOptions) (*JSInfo, error) {
 		opts.Accounts = true
 	}
 
-	// Check if we want a response from the leader only.
-	if opts.LeaderOnly {
-		js, cc := s.getJetStreamCluster()
-		if js == nil {
-			// Ignore
-			return nil, fmt.Errorf("%w: no cluster", errSkipZreq)
-		}
-		// So if we have JS but no clustering, we are the leader so allow.
-		if cc != nil {
-			js.mu.RLock()
-			isLeader := cc.isLeader()
-			js.mu.RUnlock()
-			if !isLeader {
-				return nil, fmt.Errorf("%w: not leader", errSkipZreq)
-			}
-		}
+	js, cc := s.getJetStreamCluster()
```

Comment:
Maybe move this block after line 2637? What I am not sure is, if `!s.JetStreamEnabled()` condition down there would cover this check that you did here or not.. if so, the the new error that you send here is not needed.

Reply:
Or, to avoid "races", I would replace the call to `if !s.JetStreamEnabled() {` with:
```
	jsi := &JSInfo{
		ID:  s.ID(),
		Now: time.Now().UTC(),
	}
	js, cc := s.getJetStreamCluster()
	if js == nil {
		jsi.Disabled = true
		return jsi, nil
	}

	// if we have JS but no clustering, we are the leader so allow by default
	isLeader := true
	if cc != nil {
...
```

Reply:
Yeah seems so, lets see if they pass now for sure a bug in how i refactored here

Reply:
Oh, didnt see your comment about the race before I pushed - but I guess it had that race before also - will check.

Reply:
And by race I meant that if we have `if !s.JetStreamEnabled()` and then few lines later call `s.getJetStreamCluster()`, then at that time, js may be returned as nil (if it was disabled in the meantime), so then you would still have to check again if js is nil or not. So to save on that, I would have replaced s.JetStreamEnabled() with above code: two birds in one stone.

Reply:
Pushed a change for the race thing - we already called getJetStreamCluster() earlier so just chekced if its result was nil or not

Reply:
Yes, just my suggestion was to move your code change past the place where we initialize JSInfo, etc.. but yeah, it is equivalent except that if JS was to be disabled and one day we actually use LeaderOnly, the current code would return "not leader" error instead of the jsi.Disabled result.
I think it would be more clear and consistent to have the code checking for cluster/leader after the test for js being enabled or not.